base/linalg/symmetric.jl

```diff
@@ -117,8 +117,16 @@ A_mul_B!{T<:BlasComplex,S<:StridedMatrix}(C::StridedMatrix{T}, A::StridedMatrix{
 *(A::HermOrSym, B::HermOrSym) = full(A)*full(B)
 *(A::StridedMatrix, B::HermOrSym) = A*full(B)
 
-factorize(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+bkfact(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+factorize(A::HermOrSym) = bkfact(A)
```

Comment:
Should this not try `cholfact` first @andreasnoack ? Since this is the generic factorization method. Currently:

```julia
julia> A = rand(Complex128, 4, 4); A = A'A; H = Hermitian(A);

julia> typeof(factorize(A))
Base.LinAlg.Cholesky{Complex{Float64},Array{Complex{Float64},2}}

julia> typeof(factorize(H))
Base.LinAlg.BunchKaufman{Complex{Float64},Array{Complex{Float64},2}}
```

and then change [this](https://github.com/JuliaLang/julia/blob/7affd0ced06c3596680a7ad69081aa7b98ba6ef9/base/linalg/dense.jl#L775) and [this](https://github.com/JuliaLang/julia/blob/7affd0ced06c3596680a7ad69081aa7b98ba6ef9/base/linalg/dense.jl#L779) to `bkfact(Hermitian(A))` and `bkfact(Symmetric(A))` ?